listenbrainz/labs_api/labs/api/artist_country_from_artist_credit_id.py

```diff
@@ -0,0 +1,107 @@
+from operator import itemgetter
+
+from datasethoster import Query
+import psycopg2
+import psycopg2.extras
+from werkzeug.exceptions import NotFound
+from listenbrainz import config
+
+
+class ArtistCountryFromArtistCreditIdQuery(Query):
+
+    def names(self):
+        return ("artist-country-code-from-artist-credit-id", "MusicBrainz Artist Country From Artist Credit Id")
+
+    def inputs(self):
+        return ['[artist_credit_id]']
+
+    def introduction(self):
+        return """Given artist credit id look up countries for those artists. Any artist_mbids
+                  not found in the database will be omitted from the results. If none are
+                  found a 404 error is returned."""
+
+    def outputs(self):
+        return ['artist_credit_id', 'artist_mbid', 'country_code']
+
+    def fetch(self, params, count=-1, offset=-1):
+
+        with psycopg2.connect(config.MB_DATABASE_URI) as conn:
+            with conn.cursor(cursor_factory=psycopg2.extras.DictCursor) as curs:
+
+                acs = tuple([r['[artist_credit_id]'] for r in params])
+                query = """ SELECT a.gid AS artist_mbid,
+                                    ar.id AS area_id,
+                                    code AS country_code,
+                                    ac.id AS artist_credit_id
+                               FROM artist a
+                               JOIN artist_credit_name acn
+                                 ON acn.artist = a.id
+                               JOIN artist_credit ac
+                                 ON acn.artist_credit = ac.id
+                               JOIN area ar
+                                 ON a.area = ar.id
+                    FULL OUTER JOIN iso_3166_1 iso
+                                 ON iso.area = ar.id
+                              WHERE ac.id IN %s
+                           ORDER BY artist_credit_id"""
```

Comment:
if we have LIMIT and COUNT, then this ORDER BY needs something else to order by when the AC is the same - `ORDER BY artist_credit_id, artist.id` perhaps?